MapboxVision/src/test/java/com/mapbox/vision/dsl/VisionTestDsl.kt

```diff
@@ -0,0 +1,57 @@
+package com.mapbox.vision.dsl
+
+import org.junit.jupiter.api.DynamicContainer
+import org.junit.jupiter.api.DynamicTest
+
+@DslMarker
+annotation class VisionDslMarker
+
+object TestCase {
+
+    private val givenBlocks = mutableListOf<DynamicContainer>()
+    private val whenBlocks = mutableListOf<DynamicContainer>()
+    private val thenBlocks = mutableListOf<DynamicTest>()
+
+    operator fun invoke(block: TestContext.() -> Unit): List<DynamicContainer> {
+        givenBlocks.clear()
+        TestContext().run(block)
+        return givenBlocks
+    }
+
+    internal fun addGivenBlock(displayName: String) {
```

Comment:
displayName все должны быть разные у всех тестов?

Reply:
То есть все Given в TestCase, When в Given, и Then в When

Reply:
Developer set displayName, it can be dynamic. DSL only add "Given", "When", "Then"
Right now - yes, but we can easy add block Then in Given section. 
Also I plan to add Verify function

Reply:
Ooops, I've switchied to russian for some reason

Reply:
What I meant here is - can we create more than one When inside of Given with the same name? Say, 
```
Given("...") {
  When("smth") { ... }

  When("smth") { ... }
} 
```

It seems to be this display name should be different inside of the same DynamicContainer, am i right?

Reply:
It's okay for dynamicContainer. I can add own check for that case

Reply:
Yes, seems it's not very correct case to add the same displayNames, no way to distinguish them afterwards, lets just fail if the names are equal?

Reply:
maybe add such check for v2 of DSL?

Reply:
No problem, just add issue then pls

Reply:
#174